BUILD.gn

```diff
@@ -582,9 +582,7 @@ source_set("electron_lib") {
   }
 
   if (enable_desktop_capturer) {
-    if (is_component_build && is_win) {
-      # On windows the implementation relies on unexported
-      # DxgiDuplicatorController class.
+    if (is_component_build && !is_linux) {
       deps += [ "//third_party/webrtc/modules/desktop_capture" ]
```

Comment:
What changed to add this dependency on mac ? Also please retain the comment for windows.

Reply:
@deepak1556 my bad on the comment.  As far as the mac dependency, it was need on mac to resolve this error on the debug build:
```Undefined symbols for architecture x86_64: "webrtc::GetWindowOwnerPid(unsigned int)" error.```   I can put up another PR correcting the comment.

Reply:
@deepak1556 PR to update comment is here: https://github.com/electron/electron/pull/19848

Reply:
I don't see any relevant changes on the desktop capturer api code path for these new undefined symbols. Do you know what needed these ?

Reply:
Ignore my comment, found https://github.com/electron/electron/pull/18926

Reply:
@jkleinsc I would suggest to guard those implementations added in the above PR behind `enable_desktop_capturer` since they are dependent on desktop capturer. When building with `enable_desktop_capturer=false` , things will compile horribly.

```
void TopLevelWindow::MoveAbove(const std::string& sourceId,
                               mate::Arguments* args) {
#if  BUILDFLAG(ENABLE_DESKTOP_CAPTURER)
  if (!window_->MoveAbove(sourceId))
    args->ThrowError("Invalid media source id");
#endif
}
```

Reply:
Ok I'll add it to #19848